model/policy.go

```diff
@@ -144,28 +163,31 @@ func (model Model) HasPolicies(sec string, ptype string, rules [][]string) bool
 
 // AddPolicy adds a policy rule to the model.
 func (model Model) AddPolicy(sec string, ptype string, rule []string) {
-	assertion := model[sec][ptype]
-	assertion.Policy = append(assertion.Policy, rule)
-
-	if sec == "p" && assertion.priorityIndex >= 0 {
-		if idxInsert, err := strconv.Atoi(rule[assertion.priorityIndex]); err == nil {
-			i := len(assertion.Policy) - 1
-			for ; i > 0; i-- {
-				idx, err := strconv.Atoi(assertion.Policy[i-1][assertion.priorityIndex])
-				if err != nil {
-					break
-				}
-				if idx > idxInsert {
-					assertion.Policy[i] = assertion.Policy[i-1]
-				} else {
-					break
+	if _, ok := model[sec]; ok {
```

Comment:
@arigon I'm not sure it's good to silently mute the non-existing keys, because we don't return error in this API and the caller has no mean to know that some sec or ptype doesn't work.

The caller has the reponsibility to make sure the input sec and ptype are working in his scenario (model). Or we will generate Go panic to alert him.

Or do you have a scenario for this change?

Reply:
The problem is in fact that the API is not returning an error. Generating a panic is against best practice.

See https://golang.org/doc/effective_go#panic 
> This is only an example but real library functions should avoid panic. If the problem can be masked or worked around, it's always better to let things continue to run rather than taking down the whole program.

So this should only be a temporary solution to prevent panics at all when no policy is loaded. The final solution should include returning an error but it might be a breaking change.